pipeline/src/main/java/edu/rosehulman/aixprize/pipeline/annotators/ColorsAnnotator.java

```diff
@@ -55,7 +55,16 @@ private void transmitCas(JCas cas) throws IOException {
 		StringWriter serialized = new StringWriter();
 		JsonCasSerializer.jsonSerialize(cas.getCas(), serialized);
 
-		socket.getOutputStream().write(serialized.toString().getBytes());
+		byte[] casBytes = serialized.toString().getBytes();
```

Comment:
This and the next few lines are really gross. We need a way on the Python side to tell when the JSON has ended and we can stop reading from the socket. Using a higher-level protocol like HTTP would handle things like this for us - something to consider.